### PR TITLE
feat(tsconfig): add lib option support

### PR DESCRIPTION
**Pull request by heruan**
This adds support for TypeScript 2 `lib` option in `tsconfig.json`, which replaces `noLib` and let choose which built-in libraries use to compile. E.g.:
```json
{
  "compilerOptions": {
    "rootDir": "src",
    "outDir": "dist",
    "target": "es5",
    "lib": ["es2015", "dom"],
    "declaration": true
  }
}

```

### CODE REVIEW

File: lib/main/lang/core/languageServiceHost2.ts [hunk 1]
```diff
@@ -233,9 +233,15 @@ export class LanguageServiceHost implements ts.LanguageServiceHost {
 
     constructor(private config: tsconfig.TypeScriptProjectFileDetails) {
         // Add the `lib.d.ts`
-        if (!config.project.compilerOptions.noLib) {
+        if (!config.project.compilerOptions.lib) {
           this.addScript(getDefaultLibFilePath(config.project.compilerOptions));
```
Comment (basarat) [inline lib/main/lang/core/languageServiceHost2.ts]: Please still preserve the `noLib` support :rose:

**Comment by basarat** [comment]
> which replaces noLib

doesn't replace it. It complements it. The logic should look something like it does in [alm.tools](https://github.com/alm-tools/alm/blob/5a65a9bb29a59c1b74074a59d397d78f98f22673/src/server/workers/lang/core/typeScriptDir.ts#L18-L27)  :rose:

**Comment by heruan** [comment]
Thank you @basarat I've restored the use of `noLib`. This PR is just preliminary for now, as you understand, and obviously not ready to merge. Main problem is: it does not work without also having (n)TS 2 as dependency, which breaks many other part of the plugin. I'm still testing this PR from my fork setting the TS 2.0 `typescriptServices.js` in the plugin settings.

**Comment by basarat** [comment]
⚠️ merge conflicts need fixing ⚠️  _sorry if this pr was ready and we didn't merge it before the conflicts_ :rose:

**Comment by heruan** [comment]
Thank you @basarat I've merged with upstream and fixed the conflicts!

**Comment by basarat** [comment]
Released as `10.1.2` :rose:

**Comment by heruan** [comment]
@basarat the PR didn't include the compiled `dist` and `grammars` (I'm not use to include compiled files in PRs to keep them clean). Did you compile before releasing `10.1.2`?

**Comment by basarat** [comment]
> he PR didn't include the compiled dist and grammars (I'm not use to include compiled files in PRs to keep them clean). Did you compile before releasing 10.1.2

Sorry no (my bad. Just created https://github.com/TypeStrong/atom-typescript/blob/master/pull_request_template.md to help others). Can you PR those please :rose:

**Comment by heruan** [comment]
Thank you very much! I didn't know that about Atom. https://github.com/TypeStrong/atom-typescript/pull/1012

**Pull request merged**